lte/gateway/c/session_manager/SessionStateEnforcer.h

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Magma Authors.
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*****************************************************************************
+  Source      	SessionStateEnforcer.h
+  Version     	0.1
+  Date       	2020/08/08
+  Product     	SessionD
+  Subsystem   	5G managing & maintaining state & store of session of SessionD
+                Fanout message to Access and UPF through respective client obj
+  Author/Editor Sanjay Kumar Ojha
+  Description 	Objects run in main thread context invoked by folly event
+*****************************************************************************/
+
+#pragma once
+
+#include <unordered_map>
+#include <map>
+#include <unordered_set>
+#include <vector>
+
+#include <folly/io/async/EventBaseManager.h>
+#include <lte/protos/mconfig/mconfigs.pb.h>
+#include <lte/protos/policydb.pb.h>
+#include "RuleStore.h"
+#include "PipelinedClient.h"
+#include "SessionState.h"
+#include "SessionStore.h"
+#include "AmfServiceClient.h"
+
+
+namespace magma
+{
+class SmSessionContextSendClient {};

```

Comment:
Not sure how this will be used, but if it is a separate client, it probably belongs in another file.

Reply:
Will clean up as response on the request not handled.